tools/benchmark/src/main/java/com/datadog/benchmark/internal/DatadogHttpClient.kt

```diff
@@ -22,36 +24,43 @@ import java.util.UUID
 import java.util.concurrent.TimeUnit
 
 internal class DatadogHttpClient(
-    private val context: MetricContext,
-    private val exporterConfiguration: DatadogExporterConfiguration
-) {
-
-    private val okHttpClient: OkHttpClient = OkHttpClient.Builder()
+    private val context: BenchmarkContext,
+    private val exporterConfiguration: DatadogExporterConfiguration,
+    private val callFactory: Call.Factory = OkHttpClient.Builder()
         .callTimeout(TIMEOUT_SEC, TimeUnit.SECONDS)
         .connectTimeout(TIMEOUT_SEC, TimeUnit.SECONDS)
         .readTimeout(TIMEOUT_SEC, TimeUnit.SECONDS)
         .writeTimeout(TIMEOUT_SEC, TimeUnit.SECONDS)
         .cache(null)
-        .build()
+        .build(),
+    private val metricRequestBodyBuilder: MetricRequestBodyBuilder = MetricRequestBodyBuilder(context),
+    private val spanRequestBuilder: SpanRequestBodyBuilder = SpanRequestBodyBuilder(context)
+) {
 
-    private val metricRequestBodyBuilder = MetricRequestBodyBuilder(context)
+    internal fun uploadMetric(metrics: List<MetricData>) {
+        postRequest(
+            operationName = OPERATION_NAME_METRICS,
+            url = EndPoint.AP1.metricUrl(),
```

Comment:
why do we uploading to `AP1` and not to `US1`?

Reply:
Nice catch, actually it should be neither `AP1` nor `US1`, it should be the end point defined in the configuration.